commitizen/tags.py

```diff
@@ -0,0 +1,201 @@
+from __future__ import annotations
+
+import re
+from collections.abc import Sequence
+from dataclasses import dataclass, field
+from functools import cached_property
+from string import Template
+from typing import TYPE_CHECKING
+
+from typing_extensions import Self
+
+from commitizen import out
+from commitizen.defaults import DEFAULT_SETTINGS, Settings, get_tag_regexes
+from commitizen.git import GitTag
+from commitizen.version_schemes import (
+    DEFAULT_SCHEME,
+    InvalidVersion,
+    Version,
+    VersionScheme,
+    get_version_scheme,
+)
+
+if TYPE_CHECKING:
+    from commitizen.version_schemes import VersionScheme
+
+
+@dataclass
+class TagRules:
+    """
+    Encapsulate tag-related rules
+    """
+
+    scheme: VersionScheme = DEFAULT_SCHEME
+    tag_format: str = DEFAULT_SETTINGS["tag_format"]
+    legacy_tag_formats: Sequence[str] = field(default_factory=list)
+    ignored_tag_formats: Sequence[str] = field(default_factory=list)
+    merge_prereleases: bool = False
+
+    @cached_property
+    def version_regexes(self) -> Sequence[re.Pattern]:
+        tag_formats = [self.tag_format, *self.legacy_tag_formats]
+        regexes = (self._format_regex(p) for p in tag_formats)
+        return [re.compile(r) for r in regexes]
+
+    @cached_property
+    def ignored_regexes(self) -> Sequence[re.Pattern]:
+        regexes = (self._format_regex(p, star=True) for p in self.ignored_tag_formats)
+        return [re.compile(r) for r in regexes]
+
+    def _format_regex(self, tag_pattern: str, star: bool = False) -> str:
```

Comment:
What does `star` mean? Could we please add a docstring here? Thanks!

Reply:
I'll add the missing docstrings 👍🏼